### ISSUE TITLE
Inconsistent JSON Representation of `Thing` Type from SurrealDB SDK

### ISSUE THREAD

**Issue by retrokiller543** [DV1512/backend#8]
# Description

The SurrealDB SDK uses a type called `Thing` to represent a combination of a table (`tb`) and an ID (`id`). This Thing type is necessary for parsing data into valid Rust structs. However, this results in an inconsistent JSON representation when returned to the user, especially when working with frontend applications.

# Expected JSON Response

```json
{
  "id": {
    "tb": "user",
    "id": "5f4d0c8f-1b78-4e3f-9d0c-0b0d0b0b0b0b"
  },
  "email": "johndoe@example.com",
  "url_safe_username": "johndoe",
  "username": "John Doe",
  "first_name": "John",
  "last_name": "Doe",
  "created_at": "2021-09-15T14:28:23Z",
  "last_login": "2021-09-15T14:28:23Z",
  "picture": "https://example.com/avatar.jpg",
  "role": "Owner"
}
```

# Actual JSON Response

```json
{
  "id": {
    "tb": "user",
    "id": {
      "String": "5f4d0c8f-1b78-4e3f-9d0c-0b0d0b0b0b0b"
    }
  },
  "email": "johndoe@example.com",
  "url_safe_username": "johndoe",
  "username": "John Doe",
  "first_name": "John",
  "last_name": "Doe",
  "created_at": "2021-09-15T14:28:23Z",
  "last_login": "2021-09-15T14:28:23Z",
  "picture": "https://example.com/avatar.jpg",
  "role": "Owner"
}
```

# Why This Is a Problem

This discrepancy may not significantly impact the backend, but it introduces complications for frontend applications, especially when using TypeScript with WASM-based SDKs (e.g., with `wasm-bindgen`). The nested `String` type makes it cumbersome to handle and parse the ID correctly, leading to potential type mismatches or extra parsing logic.

# Impact

This issue affects every struct that includes the `Thing` type, making it a widespread concern that could lead to inconsistent data handling across the application.

# Proposed Solution

We need to create a new response Data Transfer Object (DTO) instead of reusing the actual database DTO. This approach will allow us to control the JSON representation returned to the user and ensure consistency.

Alternatively, we should document this behavior clearly so that developers are aware of this potential idiosyncrasy when working with the SDK.

**Comment by writemorecode** [comment]
A relevant issue in the SurrealDB repo: https://github.com/surrealdb/surrealdb/issues/1609. The PR was closed with the PR https://github.com/surrealdb/surrealdb/pull/1659.

**Comment by writemorecode** [comment]
A smaller example without SurrealDB using only serde and serde_json:

```
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
enum ID {
    NumberFoo(u64),
    StringBar(String),
}

#[derive(Serialize, Deserialize, Debug)]
struct Thing {
    id: ID,
}

fn main() {
    let y = Thing {
        id: ID::NumberFoo(123_u64),
    };
    let json = serde_json::to_string_pretty(&y).unwrap();
    dbg!(y);
    println!("Record JSON: {}\n", json);

    let y = Thing {
        id: ID::StringBar(String::from("record ID")),
    };
    let json = serde_json::to_string_pretty(&y).unwrap();
    dbg!(y);
    println!("Record JSON: {}\n", json);
}
```

This example shows that the problem has to do with how serde serializes enum types to JSON.

**Comment by writemorecode** [comment]
Reading the serde documentation, I found this: https://serde.rs/enum-representations.html#untagged

Adding `#[serde(untagged)]` to the `ID` enum in the code snippet above, we get the following JSON data:

```
{
  "id": "record ID"
}
```

**Comment by retrokiller543** [comment]
manually merged with #4 to resolve conflicts
